examples/features/nse-composition/README.md

```diff
@@ -0,0 +1,286 @@
+# Test NSE composition
+
+This example demonstrates a more complex Network Service, where we chain three passthrough and one ACL Filtering NS endpoints.
+It demonstrates how NSM allows for service composition (chaining).
+It involves a combination of kernel and memif mechanisms, as well as VPP enabled endpoints.
+
+## Requires
+
+Make sure that you have completed steps from [basic](../../basic) or [memory](../../memory) setup.
+
+## Run
+
+Create test namespace:
+```bash
+NAMESPACE=($(kubectl create -f ../namespace.yaml)[0])
+NAMESPACE=${NAMESPACE:10}
+```
+
+Register namespace in `spire` server:
+```bash
+kubectl exec -n spire spire-server-0 -- \
+/opt/spire/bin/spire-server entry create \
+-spiffeID spiffe://example.org/ns/${NAMESPACE}/sa/default \
+-parentID spiffe://example.org/ns/spire/sa/spire-agent \
+-selector k8s:ns:${NAMESPACE} \
+-selector k8s:sa:default
+```
+
+Select node to deploy NSC and NSE:
+```bash
+NODE=($(kubectl get nodes -o go-template='{{range .items}}{{ if not .spec.taints  }}{{index .metadata.labels "kubernetes.io/hostname"}} {{end}}{{end}}')[0])
+```
+
+Create Passthrough and firewall NSE configurations:
+```bash
+PASS_COUNT=4
+PASS_CFG=""
+NAME=""
+LABELS=""
+VOLUME_PATCH=""
+
+for ((i = 1; i <= PASS_COUNT; i++))
+do
+  f="passthrough-${i}"
+  NAME="nse-passthrough-${i}"
+  LABELS="app:passthrough-${i}"
+  if ((i == PASS_COUNT))
+  then
+    f="nse-firewall"
+    NAME="acl-filter"
+    LABELS="app:firewall"
+    VOLUME_PATCH="- config-patch.yaml"
+  fi
+  if [ -d "${f}" ]
+  then
+    rm -r "${f}"
+  fi
+  mkdir "${f}"
+  cd "${f}" || exit
```

Comment:
Please, don't use `exit` call in integration tests. Out testing system can't handle them correctly.
You can define a shell function and use `return` statements there.

Reply:
done